kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -311,29 +478,79 @@ public void initiateConnect(HostPort remote, List<FilterAndInvoker> filters) {
             pipeline.addFirst("networkLogger", new LoggingHandler("io.kroxylicious.proxy.internal.UpstreamNetworkLogger"));
         }
 
-        virtualCluster.getUpstreamSslContext().ifPresent(c -> pipeline.addFirst("ssl", c.newHandler(outboundChannel.alloc(), remote.host(), remote.port())));
-
-        connectFuture.addListener(future -> {
+        virtualCluster.getUpstreamSslContext().ifPresent(c -> {
+            final SslHandler handler = c.newHandler(outboundChannel.alloc(), remote.host(), remote.port());
+            pipeline.addFirst("ssl", handler);
+            sslHandshakeFuture.set(handler.handshakeFuture());
+            registerExceptionResponse(SSLHandshakeException.class, throwable -> {
+                Object result;
+                final Object triggerMsg = bufferedMsgs != null ? bufferedMsgs.get(0) : null;
+                if (triggerMsg instanceof final DecodedRequestFrame<?> triggerFrame) {
+                    result = buildErrorResponseFrame(triggerFrame, new UnknownServerException());
```

Comment:
What would you think to using NetworkException rather than UnknownServerException?   At least that hints to the transporty nature of the issue.   I considered `BrokerNotAvailableException` but ruled this out as potentially misleading.

Reply:
`BrokerNotAvailableException` was initially alluring to me.  Then I looked at how the error is used in Kafka itself - it seems to used in cases of   transaction failures or quorum issues.    Putting myself in the shoes of examining a client application log googling the error, I might jump to the wrong conclusion about the nature of the error and send the Kafka broker team looking in the wrong place.

At least `NetworkException` screams connectivity issue.

Reply:
that's a fair call. I'll update